From 848e9a668a23e9a555754de8942622c6f0b90fa6 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Thu, 1 Feb 2024 15:19:37 +0100
Subject: [PATCH] dt-bindings: dma: stm32-dma3: introduce memory-region
 property

DMA3 requires memory to allocate its linked-list items (LLIs). By default,
when memory-region property is not specified, LLIs are allocated in DDR.
Thanks to memory-region property, a dedicated /reserved-memory child node
can be used for LLIs allocation. It is then recommended to use internal
memory: LLI read for registers update is optimal from internal RAM.

Change-Id: Ifcbb90a902bb9be99c9ef740c93df4e2595c3f9f
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/370775
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 .../devicetree/bindings/dma/stm32/st,stm32-dma3.yaml     | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/Documentation/devicetree/bindings/dma/stm32/st,stm32-dma3.yaml b/Documentation/devicetree/bindings/dma/stm32/st,stm32-dma3.yaml
index 8d10f0962260..71147e6af4d8 100644
--- a/Documentation/devicetree/bindings/dma/stm32/st,stm32-dma3.yaml
+++ b/Documentation/devicetree/bindings/dma/stm32/st,stm32-dma3.yaml
@@ -48,6 +48,15 @@ properties:
   resets:
     maxItems: 1
 
+  memory-region:
+    description: |
+      Phandle to a /reserved-memory child node assigned to STM32 DMA3 for linked-list items
+      allocation. When not defined, linked-list items are allocated in DDR.
+      During transfer, STM32 DMA3 updates its registers reading the linked-list items in memory.
+      There reads are 32-bit single read. Allocating the linked-list items in internal RAM improves
+      the STM32 DMA3 performance.
+    maxItems: 1
+
   power-domains:
     maxItems: 1
 
-- 
2.39.5

